### DISCUSSION TITLE
请问如何将输入编码“123a”转换成“1a2a3a”？

### DISCUSSION THREAD

**Discussion by cloudfish** [rime/home#1328] [Q&A] [answered]
是这样的，我想写一个输入方案，经过简化以后，需要解决这样一个问题：
假设我需要输入各种不同形式的数字，例如在码表中，1a表示①，2a表示 ②，1b表示㊀，2b表示㊁，以此类推。那么输入1a2a3a，候选就应当是① ② ③。但是同时我还希望支持缩写形式，即123a输出① ② ③，12a12b输出①②㊀㊁。我读了一下拼写运算的文档，因为我对正则表达式也不是很熟悉，想不到怎么用xform实现这个，因为数字可以有任意个。如果先用segmentor把12a和12b分开处理的话，那对应的xform就是在每个数字后面添加a或者b就行了，但是不知道segmentor支不支持这样的操作。
请问大家有没有什么好的思路？

**Comment by lotem** [comment, marked as answer]
我的思路是，先用 `recognizer/patterns` 定義 `[0-9]+a` 和 `[0-9]+b` 兩種輸入模式，分列兩個 translator 翻譯有對應標記類型的 segment，而單字的輸入碼可以省略後綴 `a`、`b`。
